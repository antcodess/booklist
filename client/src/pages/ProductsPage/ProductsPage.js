import React from 'react';
import { Route, Switch } from 'react-router-dom';

import ProductItem from '../../components/ProductItem/ProductItem';
import ShopPageContainer from '../ShopPage/ShopPageContainer';

import './ProductPage.scss';


const ProductsPage = ({ match }) => {
  return (
    <Switch>
      <Route path={`${match.path}/:productTitle`} component={ProductItem} />
      <Route path={`${match.path}`} component={ShopPageContainer} />
    </Switch>
  );
};

export default ProductsPage;