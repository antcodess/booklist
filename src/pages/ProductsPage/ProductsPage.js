import React from 'react';
import { Route, Switch } from 'react-router-dom';

import ProductItem from '../../components/ProductItem/ProductItem';
import ProductsOverview from '../../components/ProductsOverview/ProductsOverview';
import './ProductPage.scss';

const ProductsPage = ({ match }) => {
  return (
    <Switch>
      <Route path={`${match.path}/:productTitle`} component={ProductItem} />
      <Route path={`${match.path}`} component={ProductsOverview} />
    </Switch>
  );
};

export default ProductsPage;