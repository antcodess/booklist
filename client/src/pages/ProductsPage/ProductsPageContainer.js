import { connect } from 'react-redux';
import { compose } from 'redux';
import { createStructuredSelector } from 'reselect';
import { selectIsProductLoaded } from '../../redux/shop/shop.selectors';

import WithSpinner from '../../components/WithSpinner/WithSpinner';
import ProductsPage from './ProductsPage';

const mapStateToProps = createStructuredSelector({
  isLoading: state => !selectIsProductLoaded(state)
});

const ProductPageContainer = compose(
  connect(mapStateToProps),
  WithSpinner
)(ProductsPage);

export default ProductPageContainer;

