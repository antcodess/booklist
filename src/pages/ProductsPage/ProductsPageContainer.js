import { connect } from 'react-redux';
import { compose } from 'redux';
import { createStructuredSelector } from 'reselect';
import { selectIsProductLoaded } from '../../redux/shop/shop.selectors';

import Spinner from '../../components/Spinner/Spinner';
import ProductsPage from './ProductsPage';

const mapStateToProps = createStructuredSelector({
  isLoading: state => !selectIsProductLoaded(state)
});

const ProductPageContainer = compose(
  connect(mapStateToProps),
  Spinner
)(ProductsPage);

export default ProductPageContainer;

