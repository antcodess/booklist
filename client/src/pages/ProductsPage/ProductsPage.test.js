import React from 'react';
import { shallow } from 'enzyme';
import { ProductsPage } from './ProductsPage';

describe('ProductsPage Component', () => {
  let wrapper;
  let mockMatch;

  beforeEach(() => {
    mockMatch = {
      path: '/products'
    }

    const mockProps = {
      match: mockMatch
    };
    wrapper = shallow(<ProductsPage { ...mockProps } />);
  });

  it('should render ProductsPage Component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});