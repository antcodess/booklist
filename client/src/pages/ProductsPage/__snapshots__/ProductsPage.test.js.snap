// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsPage Component should render ProductsPage Component 1`] = `ShallowWrapper {}`;
